weights.py

```diff
@@ -189,19 +220,45 @@ def download_safetensors(url: str, path: Path):
         raise RuntimeError(f"Failed to download safetensors file: {e}")
 
 
-def make_download_url(url: str) -> str:
+def make_download_url(
+    url: str, hf_api_token: str | None = None, civitai_api_token: str | None = None
+) -> str:
     if url.startswith("data:"):
         return url
-    if m := re.match(r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)/?$", url):
-        owner, model_name = m.groups()
-        return make_huggingface_download_url(owner, model_name)
+    if m := re.match(
+        r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)(?:/([^/]+\.safetensors))?/?$",
+        url,
+    ):
+        if len(m.groups()) != 3 or not m.groups()[2]:
+            raise ValueError(
+                "Invalid HuggingFace URL. Expected format: huggingface.co/<owner>/<model-name>/<lora-weights-file.safetensors>"
+            )
+
```

Comment:
agreed that it's a bit wonky having this code in the `make_download_url` function. I'd argue that we can treat this as a separate branch in `download_weights` - e.g. in the `if hf_api_token is not None and 'huggingface.co' in url:` clause we just call some other `hf_download` function.